recipes/apr/all/conanfile.py

```diff
@@ -83,6 +89,56 @@ def build_requirements(self):
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
+    def _get_cross_building_configure_args(self):
+        """
+        The APR configure.in script makes extensive (30 instances) use
+        Check warning on line 73 in recipes/apr/all/conanfile.py
+
+        String statement has no effect of AC_TRY_RUN (to determine system capabilities) and
+        checks for /dev/zero. These runtime checks only work when run on the host so
+        the configure script will fail when cross compiling unless the relevant
+        configuration variables are provided in a cache file or on the command line or in an
+        environment variable. The configuration variable values are most easily determined by
+        running the configure script on a host system using:
+
+            ./configure --cache-file={gnu_host_triplet}.cache
+
+        The generated cache file can be repeatedly used to cross-compile to the targeted host system
+        by including it with the recipe data.
+
+        This recipe is reading this custom user conf variable:
+
+            [conf]
+            user.apr:cache_file=/path/to/{gnu_host_triplet}.cache
+
+        So you can use it to cross-compile on your system.
+        """
+        configure_args = []
+        user_cache_file = self.conf.get("user.apr:cache_file", check_type=str)
+        if user_cache_file:
+            configure_args.append(f"--cache-file={user_cache_file}")
+            return configure_args
+
+        self.output.warning("Trying to set some configuration arguments, but it"
+                            " could fail. The best approach is to provide a"
+                            " pre-built cached file.")
+        # Let's try this hack (tested on Linux ARM and Intel)
```

Comment:
not sure I'd call this a hack,  I think passing these variables to the configure script is valid and for example is documented in the changelog for this very purpose, but i could be wrong:
https://github.com/apache/apr/commit/d484a5ba10d4ea4e623b3e3969d3fdf54791e7e6